authenticate.js

```diff
@@ -20,21 +20,30 @@ const OPTIONS = {
 
 const loginAttempt = () => {
     return new Promise((resolve, _reject) => {
-        if (!clientId || !secret) {
-            alfy.error('OAuth2 not set. Refer to README for details');
-        } else {
-            const accessToken = alfy.cache.get(ACCESS_TOKEN);
-            if (accessToken) {
-                resolve(bitbucket(accessToken));
-            } else {
-                alfy.fetch(URL, OPTIONS).then(({ access_token }) => {
-                    alfy.cache.set(ACCESS_TOKEN, access_token, { maxAge: ALIVE_TIME });
-                    resolve(bitbucket(access_token));
-                }).catch(() => {
-                    alfy.cache.set(ACCESS_TOKEN, null);
-                });
-            }
+        if ((!username || !appPassword) && (!clientId || !secret)) {
+            return alfy.error('OAuth2 not set. Refer to README for details');
+        }
+
+        if (username && appPassword) {
+            return resolve(bitbucket(`Basic ${Buffer.from([username, appPassword].join(':')).toString('base64')}`));
         }
+
+        const accessToken = alfy.cache.get(ACCESS_TOKEN);
+
+        if (accessToken) {
+            return resolve(bitbucket(accessToken));
+        }
```

Comment:
Nice work! Since both are given as the same argument like the accessToken, I think I would put that in a function where the app password check is done. This function is getting large too I think.

So perhaps something like 
```js
const getAccessToken = () => {
  if (username && appPassword) {
    return `Basic ${Buffer.from([username, appPassword].join(':')).toString('base64')}`
  }

  if (accessToken) {
    return accessToken
  }

  return null
}
```

(that's pseudo-js, just as example)

Tell me what you think!

Reply:
I was trying to be as least-invasive as possible. I did some refactoring, in the style I would think is best. you can let me know if you disagree and I can revert it.